theano/tensor/nnet/nnet.py

```diff
@@ -384,6 +384,7 @@ def c_code(self, node, name, inp, out, sub):
             }
         }
 
+        #pragma omp parallel for schedule(static, 32)
```

Comment:
thanks. I did some local benchmark. Using openmp=True, vs openmp=False make a speed difference. But here, openmp=True is slower in a small test case:

```
        x = T.matrix()
        f = theano.function([x], softmax_op(x))

        xv = numpy.random.randn(600, 12800).astype(config.floatX)
        for i in range(10):
            f(xv)
```

What shapes did you used? If the problem is that my shapes are too small, I think we should a condition to the openmp pragma to not run for too small shapes. We do this for elemwise too.

Note, we also have the ops SoftmaxWithBias that is used sometimes. When we have softmax working well, doing the same for it would be great.